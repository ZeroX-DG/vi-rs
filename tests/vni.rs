mod shared;

fn snapshot_transform(lines: &str) -> String {
    shared::transform_lines(lines, |word| {
        let mut trasformed_word = String::new();
        vi::transform_buffer(&vi::VNI, word.chars(), &mut trasformed_word);
        trasformed_word
    })
}

macro_rules! gen_test_vi {
    ($name:tt, $path: tt) => {
        gen_test!(snapshot_transform, $name, $path);
    };
}
gen_test_vi!(simple_vni, "../testdata/input/simple_vni.txt");
gen_test_vi!(
    non_vietnamese_vni,
    "../testdata/input/non_vietnamese_vni.txt"
);
