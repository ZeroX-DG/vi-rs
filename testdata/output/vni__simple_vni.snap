---
source: tests/vni.rs
expression: snapshot_transform(contents)
info:
  input_file: "../testdata/input/simple_vni.txt"
---
tôi là ai
đây là đâu
thêm một cái nữa
ví tiền
chào
vơ
ru
vươn
chÊ
ă
111222333444555666777888999000
chệch
che8
vươn
a1
hoảng
gì
việt nam
xuất hiện xuất hiện xuất hiện
ha

